autokeras/hypermodel/hyperblock.py

```diff
@@ -207,7 +206,7 @@ def build_feature_engineering(self, hp, input_node):
                                             default=True)
         if feature_engineering:
             output_node = preprocessor.FeatureEngineering(
-                column_types=self.column_types)(output_node)
+                input_node=self.input_node)(output_node)
```

Comment:
Input_node should not be in the constructor but an attribute to be set after initialization.